Sources/File.swift

```diff
@@ -188,14 +188,10 @@ extension File {
         return filedetach(file)
     }
 
-    public func close() -> Bool {
-        if closed {
-            return false
-        }
-
+    public func close() throws {
+        guard !closed else { throw ClosableError.alreadyClosed }
```

Comment:
I don't think we would throw exceptions if the connection is already closed. We should reserve it for times when the stream doesn't wish to be closed yet.